src/main/java/org/broadinstitute/hellbender/tools/spark/sv/discovery/AnnotatedVariantProducer.java

```diff
@@ -29,51 +29,49 @@
      */
-    static VariantContext produceAnnotatedVcFromNovelAdjacency(final NovelAdjacencyReferenceLocations novelAdjacencyReferenceLocations,
-                                                               final SvType inferredType,
-                                                               final Iterable<ChimericAlignment> contigAlignments,
-                                                               final Broadcast<ReferenceMultiSource> broadcastReference)
+    static VariantContext produceAnnotatedVcFromInferredTypeAndRefLocations(final SimpleInterval refLoc, final int end,
+                                                                            final BreakpointComplications breakpointComplications,
+                                                                            final SvType inferredType,
+                                                                            final Iterable<ChimericAlignment> contigAlignments,
+                                                                            final Broadcast<ReferenceMultiSource> broadcastReference)
             throws IOException {
 
-        final String contig = novelAdjacencyReferenceLocations.leftJustifiedLeftRefLoc.getContig();
-        final int    start  = novelAdjacencyReferenceLocations.leftJustifiedLeftRefLoc.getEnd();
-        final int    end    = novelAdjacencyReferenceLocations.leftJustifiedRightRefLoc.getStart();
+        final int applicableEnd = end < 0 ? refLoc.getEnd() : end; // BND formatted variant shouldn't have END
 
         // basic information and attributes
         final VariantContextBuilder vcBuilder = new VariantContextBuilder()
-                .chr(contig).start(start).stop(end)
-                .alleles(produceAlleles(novelAdjacencyReferenceLocations, broadcastReference.getValue(), inferredType))
+                .chr(refLoc.getContig()).start(refLoc.getStart()).stop(applicableEnd)
+                .alleles(produceAlleles(refLoc, broadcastReference.getValue(), inferredType))
                 .id(inferredType.getInternalVariantId())
-                .attribute(VCFConstants.END_KEY, end)
                 .attribute(GATKSVVCFConstants.SVTYPE, inferredType.toString())
                 .attribute(GATKSVVCFConstants.SVLEN, inferredType.getSVLength());
 
         // attributes from complications
         inferredType.getTypeSpecificAttributes().forEach(vcBuilder::attribute);
-        parseComplicationsAndMakeThemAttributeMap(novelAdjacencyReferenceLocations.complication).forEach(vcBuilder::attribute);
+        parseComplicationsAndMakeThemAttributeMap(breakpointComplications).forEach(vcBuilder::attribute);
 
         // evidence used for producing the novel adjacency
         getEvidenceRelatedAnnotations(contigAlignments).forEach(vcBuilder::attribute);
 
-        return vcBuilder.make();
+        return end > 0 ? vcBuilder.attribute(VCFConstants.END_KEY, applicableEnd).make() : vcBuilder.make();
```

Comment:
It might be nicer to only call make once and only conditionally add the attribute in a prior line (you'd have to make the builder not be final, though).